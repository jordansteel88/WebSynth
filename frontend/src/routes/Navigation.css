.Navigation {
  background-color: #71808f;
}