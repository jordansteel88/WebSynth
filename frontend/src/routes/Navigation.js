import React, { useContext, useState } from "react";
import "./Navigation.css";
import UserContext from "../auth/UserContext";
import { NavLink } from "react-router-dom";
import {
  Collapse,
  Navbar,
  NavbarToggler,
  Nav,
  NavItem
} from 'reactstrap';


const Navigation = ({ logout }) => {
  const { currentUser } = useContext(UserContext);

  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen); 


  const loggedInNav = () => {
    let profilesLink = `/profiles/${currentUser.username}`;
    console.log(profilesLink);

    return (
      <Nav className="ml-auto">       
        <NavItem className="mr-3">
          <NavLink className="text-white" to={profilesLink}>
            {currentUser.username}'s Profiles
          </NavLink>
        </NavItem>        
        <NavItem className="mr-2">
          <NavLink className="text-white" to="/" onClick={logout}>
            Log Out
          </NavLink>
        </NavItem>
      </Nav>
    );
  }   
    
  const loggedOutNav = () => {
    return (
      <Nav className="ml-auto">
        <NavItem className="mr-3">
          <NavLink className="text-white" to="/login">
            Log In
          </NavLink>
        </NavItem>        
        <NavItem className="mr-2">
          <NavLink className="text-white" to="/register">
            Sign Up
          </NavLink>
        </NavItem>        
      </Nav>
    );
  }

  return (
    <div className="Navigation">
      <Navbar expand="md">
        <Nav>
          <NavItem className="mr-3">
            <NavLink className="text-white" to="/">
              WebSynth
            </NavLink>
          </NavItem>
        </Nav>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          {currentUser ? loggedInNav() : loggedOutNav()}           
        </Collapse>
      </Navbar>
    </div>
  );  
}

export default Navigation;