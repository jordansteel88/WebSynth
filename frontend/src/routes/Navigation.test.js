import React from "react";
import { render } from "@testing-library/react";
import Navigation from "./Navigation";
import { BrowserRouter } from "react-router-dom";
import UserContext from '../auth/UserContext';

jest.mock('../components/Synth.js', () => {
  return function DummySynth() {
    return (
      <div>DummySynth</div>
  )};
})

describe("Navigation", () => {
  const initialState = {
    currentUser: "test"
  }

  it("renders without crashing", function() {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ initialState }}>
          <Navigation />
        </UserContext.Provider> 
      </BrowserRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ initialState }}>
          <Navigation />
        </UserContext.Provider> 
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
})