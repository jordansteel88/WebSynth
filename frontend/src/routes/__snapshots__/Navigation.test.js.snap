// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Navigation"
  >
    <nav
      class="navbar navbar-expand-md"
    >
      <ul
        class="nav"
      >
        <li
          class="mr-3 nav-item"
        >
          <a
            aria-current="page"
            class="text-white active"
            href="/"
          >
            WebSynth
          </a>
        </li>
      </ul>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ml-auto nav"
        >
          <li
            class="mr-3 nav-item"
          >
            <a
              class="text-white"
              href="/login"
            >
              Log In
            </a>
          </li>
          <li
            class="mr-2 nav-item"
          >
            <a
              class="text-white"
              href="/register"
            >
              Sign Up
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
